browser-test/src/applicant/application_navigation.test.ts

```diff
@@ -49,39 +53,299 @@ test.describe('Applicant navigation flow', () => {
         'nav-date-q',
         'nav-email-q',
       ])
-      await adminPrograms.addProgramBlock(programName, 'second description', [
-        'nav-static-q',
-      ])
-      await adminPrograms.addProgramBlock(programName, 'third description', [
-        'nav-address-q',
-      ])
-      await adminPrograms.addProgramBlock(programName, 'fourth description', [
-        'nav-radio-q',
-      ])
-      await adminPrograms.addProgramBlock(programName, 'fifth description', [
-        'nav-phone-q',
-      ])
+      await adminPrograms.addProgramBlockUsingSpec(
+        programName,
+        'second description',
+        [{name: 'nav-static-q', isOptional: false}],
+      )
+      await adminPrograms.addProgramBlockUsingSpec(
+        programName,
+        'third description',
+        [{name: 'nav-address-q', isOptional: false}],
+      )
+      await adminPrograms.addProgramBlockUsingSpec(
+        programName,
+        'fourth description',
+        [{name: 'nav-radio-q', isOptional: true}],
+      )
+      await adminPrograms.addProgramBlockUsingSpec(
+        programName,
+        'fifth description',
+        [
+          {name: 'nav-phone-q', isOptional: false},
+          {name: 'nav-currency-q', isOptional: true},
+        ],
+      )
 
       await adminPrograms.gotoAdminProgramsPage()
       await adminPrograms.publishProgram(programName)
     })
 
+    test.describe('next button', () => {
+      test('next block progression', async () => {
+        const {page, applicantQuestions} = ctx
+        await applicantQuestions.clickApplyProgramButton(programName)
+
+        await validateAccessibility(page)
+        await validateScreenshot(
+          page,
+          'program-preview',
+          /* fullPage= */ true,
+          /* mobileScreenshot= */ true,
+        )
+
+        await test.step('incomplete screen 1', async () => {
+          await applicantQuestions.clickContinue()
+          await applicantQuestions.validateQuestionIsOnPage(dateQuestionText)
+          await applicantQuestions.validateQuestionIsOnPage(emailQuestionText)
+          await applicantQuestions.answerDateQuestion('2021-11-01')
+          await applicantQuestions.answerEmailQuestion('')
+          await applicantQuestions.clickNext()
+
+          // Expect we're still on the same page and see errors since email wasn't answered
+          await applicantQuestions.validateQuestionIsOnPage(dateQuestionText)
+          await applicantQuestions.validateQuestionIsOnPage(emailQuestionText)
+          expect(await page.innerText('.cf-question-email')).toContain(
+            'This question is required',
+          )
+        })
+
+        await test.step('complete screen 1', async () => {
+          await applicantQuestions.answerDateQuestion('2021-11-01')
+          await applicantQuestions.answerEmailQuestion('test1@gmail.com')
+        })
+
+        await test.step('screen 2', async () => {
+          await applicantQuestions.clickNext()
```

Comment:
nit:  I expected `clickNext()` to be part of the "complete screen 1" step since "incomplete screen 1" includes it.  Also, isn't the "next" button on screen 1?

Reply:
Good point! I was really conflicted on which step to put `clickNext` in, so your feedback is super helpful. I moved to the previous part!